addons/default/visiosoft/advs-module/src/Adv/AdvRepository.php

```diff
@@ -391,6 +391,17 @@ public function latestAds()
         return $ads;
     }
 
+	public function bestsellerAds($catId = null, $limit = 10)
+	{
+		return $this->model->orderBy('total_sales', 'desc')
+			->where(function ($query) use ($catId) {
+				if ($catId) {
+					$query->where('cat1', $catId);
+				}
+			})
+			->limit($limit)->get();
```

Comment:
maybe you need to check the status? and if the ad is expired?

Reply:
we need to use common method for it. it's why we have model and repo